esrally/utils/io.py

```diff
@@ -274,24 +285,55 @@ def decompress(zip_name, target_directory):
     this function.
     """
     path_without_extension, extension = splitext(zip_name)
-    filename = basename(path_without_extension)
     if extension == ".zip":
         _do_decompress(target_directory, zipfile.ZipFile(zip_name))
     elif extension == ".bz2":
-        _do_decompress_manually(target_directory, filename, bz2.open(zip_name))
+        decompressor_args = ["pbzip2", "-d", "-k", "-m10000", "-c"]
+        decompressor_lib = bz2.open(zip_name)
```

Comment:
We could avoid opening the file prematurely by just passing the function reference and calling it when needed? We can also do this with a context manager to ensure the compressed file is properly closed again afterwards. 

We're also not closing the open file in case the file can be decompressed via the binary instead of the library.

Reply:
Addressed in c51403c60110882237bc420eca02a9739b148072